src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -65,33 +65,58 @@ std::vector<std::string> get_supported_metrics() {
           "confusion_matrix", "roc_curve"};
 }
 
-std::tuple<float,size_t> get_accuracy(size_t prediction_window, size_t num_classes, shared_float_array output_chunk, shared_float_array classes_chunk, data_iterator::batch::chunk_info info ){
+std::tuple<float,size_t> get_accuracy_per_sequence(size_t prediction_window, size_t num_classes, 
+  const shared_float_array& output_chunk, const shared_float_array& classes_chunk, data_iterator::batch::chunk_info info) {
       
 
-    const float* output_ptr = output_chunk.data();
-    const float* truth_ptr = classes_chunk.data();
+  const float* output_ptr = output_chunk.data();
+  const float* truth_ptr = classes_chunk.data();
     
-    size_t cumulative_samples = 0;
-    size_t seq_len = 0;
-    float cumulative_seq_accuracy = 0.f;
+  size_t cumulative_samples = 0;
+  size_t num_predictions = 0;
```

Comment:
Honestly, we can compute `num_predictions` ahead of time as e.g.
```
size_t num_predictions = (info.num_samples + prediction_window - 1) / prediction_window;
```
Note that the `+ prediction_window - 1` part is just to round up instead of round down.

But then this can just be computed by the caller of this function and passed in as a parameter